src/ImageSharp/Formats/WebP/WebPLossyDecoder.cs

```diff
@@ -0,0 +1,1414 @@
+            {
+                dec.SkipProbability = (byte)this.bitReader.ReadValue(8);
+            }
+        }
+
+        private static Vp8Io InitializeVp8Io(Vp8Decoder dec, Vp8PictureHeader pictureHeader)
+        {
+            var io = default(Vp8Io);
+            io.Width = (int)pictureHeader.Width;
+            io.Height = (int)pictureHeader.Height;
+            io.UseScaling = false;
+            io.ScaledWidth = io.Width;
+            io.ScaledHeight = io.ScaledHeight;
+            io.MbW = io.Width;
+            io.MbH = io.Height;
+            io.YStride = (int)(16 * ((pictureHeader.Width + 15) >> 4));
+            io.UvStride = (int)(8 * ((pictureHeader.Width + 15) >> 4));
+
+            int intraPredModeSize = 4 * dec.MbWidth;
+            dec.IntraT = new byte[intraPredModeSize];
+
+            int extraPixels = WebPConstants.FilterExtraRows[(int)dec.Filter];
+            if (dec.Filter is LoopFilter.Complex)
+            {
+                // For complex filter, we need to preserve the dependency chain.
+                dec.TopLeftMbX = 0;
+                dec.TopLeftMbY = 0;
+            }
+            else
+            {
+                // For simple filter, we include 'extraPixels' on the other side of the boundary,
+                // since vertical or horizontal filtering of the previous macroblock can modify some abutting pixels.
+                dec.TopLeftMbX = (-extraPixels) >> 4;
+                dec.TopLeftMbY = (-extraPixels) >> 4;
+                if (dec.TopLeftMbX < 0)
+                {
+                    dec.TopLeftMbX = 0;
+                }
+
+                if (dec.TopLeftMbY < 0)
+                {
+                    dec.TopLeftMbY = 0;
+                }
+            }
+
+            // We need some 'extra' pixels on the right/bottom.
+            dec.BottomRightMbY = (io.Height + 15 + extraPixels) >> 4;
+            dec.BottomRightMbX = (io.Width + 15 + extraPixels) >> 4;
+            if (dec.BottomRightMbX > dec.MbWidth)
+            {
+                dec.BottomRightMbX = dec.MbWidth;
+            }
+
+            if (dec.BottomRightMbY > dec.MbHeight)
+            {
+                dec.BottomRightMbY = dec.MbHeight;
+            }
+
+            return io;
+        }
+
+        [MethodImpl(InliningOptions.ShortMethod)]
+        private static uint NzCodeBits(uint nzCoeffs, int nz, int dcNz)
+        {
+            nzCoeffs <<= 2;
+            nzCoeffs |= (uint)((nz > 3) ? 3 : (nz > 1) ? 2 : dcNz);
+            return nzCoeffs;
+        }
+
+        [MethodImpl(InliningOptions.ShortMethod)]
+        private static Vp8BandProbas[] GetBandsRow(Vp8BandProbas[,] bands, int rowIdx)
+        {
+            Vp8BandProbas[] bandsRow = Enumerable.Range(0, bands.GetLength(1)).Select(x => bands[rowIdx, x]).ToArray();
+            return bandsRow;
+        }
+
+        [MethodImpl(InliningOptions.ShortMethod)]
+        private static int CheckMode(int mbx, int mby, int mode)
```

Comment:
this method is used twice and the only use of the result is to switch over the values.
This get's more readable and should be equally fast (I hope) when we introduce a private enum for CheckMode.
It would allow to use self-describing values instead.